legal-api/src/legal_api/resources/v2/configuration.py

```diff
@@ -35,3 +36,114 @@ def get_configurations():
             configuration.json for configuration in configurations
         ]
     }), HTTPStatus.OK
+
+
+@bp.route('', methods=['PUT'])
+@cross_origin(origin='*')
+@jwt.has_one_of_roles([UserRoles.staff])
+def update_configurations():
+    """Update the configurations."""
+    json_input = request.get_json()
+    if not json_input:
+        return {'message': 'Request body cannot be blank'}, HTTPStatus.BAD_REQUEST
+
+    configurations = json_input.get('configurations', [])
+    error_message = validate_configurations(configurations)
+    if error_message:
+        return {'message': error_message}, HTTPStatus.BAD_REQUEST
+
+    response = []
+    try:
+        for config_data in configurations:
+            name = config_data.get('name')
+            value = config_data.get('value')
+            config = Configuration.find_by_name(name)
+            config.val = str(value)
+            db.session.add(config)
+            response.append(config.json)
+        db.session.commit()
+    except ValueError as e:
+        # Rollback transaction
+        db.session.rollback()
+        return {'message': str(e)}, HTTPStatus.BAD_REQUEST
+
+    return {'configurations': response}, HTTPStatus.OK
+
+
+def validate_configurations(configurations):
+    """Validate the configurations before updating."""
+    if not configurations:
+        return 'Configurations list cannot be empty'
+
+    # Extract names from the requested configuration updates
+    names = [config['name'] for config in configurations]
+    if len(names) != len(set(names)):
+        return 'Duplicate names error.'
```

Comment:
i think there is a scenario where duplicates won't be picked up here...not sure.  think we may want to uppercase all the entries in `names`